source/r/phyloseq-to-metabar.R

```diff
@@ -0,0 +1,276 @@
+download_and_source <- function(url, destination) {
+  if (!file.exists(destination)) {
+    download.file(url, destination, mode = "wb")
+  }
+  source(destination)
+}
+
+# Download and source microfiltR_source_code.R
+download_and_source("https://raw.githubusercontent.com/itsmisterbrown/microfiltR/master/microfiltR_source_code.R", "microfiltR_source_code.R")
+
+# Download and source check_metabarlist.R
+download_and_source("https://raw.githubusercontent.com/metabaRfactory/metabaR/HEAD/R/check_metabarlist.R", "check_metabarlist.R")
+
+# Download and source subset_metabarlist.R
+download_and_source("https://raw.githubusercontent.com/metabaRfactory/metabaR/HEAD/R/subset_metabarlist.R", "subset_metabarlist.R")
+
+# Download and source ggtaxplot.R
+download_and_source("https://raw.githubusercontent.com/slambrechts/metabaR/master/R/ggtaxplot.R", "ggtaxplot.R")
+
+################################################################################
+# Set working directory
+
+#setwd("G:/path/to/working/directory")
+
+################################################################################
+################################################################################
+# Load in Phyloseq Olig01 Annelida MBAG
+load("./physeq_Olig01_Annelida_rar_species.Rdata")
+
+
+fp <- paste(getwd(), "/", sep = "")
+write.dataset(ps = physeq_Olig01_Annelida_rar.rarefied.species, filePATH = fp, filePREFIX = "Olig01_Annelida")
+
+
+################################################################################
+# File reads
+asv_tab <- as.matrix(t(phyloseq::otu_table(physeq_Olig01_Annelida_rar.rarefied.species)))
+file_reads <- as.data.frame(asv_tab)
+
+# Make numeric columns numeric
+str(file_reads)
+
+file_reads_num <- file_reads
+file_reads_num[] <- as.data.frame(sapply(file_reads, as.numeric))
+str(file_reads_num)
+
+# create matrix
+file_reads_num_matrix <- as.matrix(file_reads_num)
+class(file_reads_num_matrix)
+
+# Add _r1 to the row names
+# adding suffix to row names
+rownames(file_reads_num_matrix) <- paste(rownames(file_reads_num_matrix),"r1",sep = "_")
+
+
+# Save file
+write.table(file_reads_num_matrix, file = "Olig01_Annelida_ASV_table_transpose.txt", sep = "\t",
+            row.names = TRUE)
+
+
+################################################################################
+#file pcrs (moet eerst nog  aangemaakt worden)
+
+
+# Assuming your data is stored in a matrix or a data frame named 'your_data'
+# First, create a dataframe with the first two columns
+df_pcr <- data.frame(
+  pcr_id = rownames(file_reads_num_matrix),
+  sample_id = sub("_r1$", "", rownames(file_reads_num_matrix)),
+  stringsAsFactors = FALSE
+)
+
+# Add the last two columns
+df_pcr$type <- "sample"
+df_pcr$control_type <- NA
+
+# create matrix
+file_pcrs_matrix <- as.matrix(df_pcr)
+class(file_pcrs_matrix)
+
+# Save file
+write.table(file_pcrs_matrix, file = "Olig01_Annelida_PCRs.txt", sep = "\t",
+            row.names = FALSE)
+
```

Comment:
in plaats van telkens objecten weg te schrijven kan het waarschijnlijk eenvoudiger via https://metabarfactory.github.io/metabaR/reference/metabarlist_generator.html waarmee rechtstreeks R objecten kunnen doorgegeven worden aan de argumenten